enhancements/authentication/login-logout-events.md

```diff
@@ -0,0 +1,226 @@
+---
+title: Login Logout Auditing
+authors:
+  - "@s-urbaniak"
+reviewers:
+  - "@mfojtik"
+  - "@deads2k"
+  - "@stlaz"
+  - "@sttts"
+approvers:
+- "@Anandnatraj"
+- "@oscardoe"
+creation-date: 2021-09-10
+last-updated: 2021-09-10
+status: implementable
+see-also:
+  - "/enhancements/kube-apiserver/audit-policy.md"
+---
+
+# Logging login, login failure and logout events
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Operational readiness criteria is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+This enhancement describes capturing login, login failure, and logout events in the OpenShift authentication subsystems.
+
+Currently, login and logout events can only be derived from inspecting audit logs. These include creation or deletion of `oauthaccesstokens` resources as described in the API server [Audit Policy Configuration](https://github.com/openshift/enhancements/blob/master/enhancements/kube-apiserver/audit-policy.md#logging-of-token-with-secure-oauth-storage).
+
+This approach has a couple of drawbacks:
+1. Login events are captured via creation events of `oauthaccesstokens` in audit logs. This includes successful login events only, but not all login failures. Login failures can be only deduced by comparing which `oauthauthorizetokens` were created and which were not exchagend for an `oauthaccesstoken`.
```

Comment:
`Login failures can be only deduced by comparing which oauthauthorizetokens were created and which were not exchagend for an oauthaccesstoken`
->
 `Currently the only login failure that can be deduced from the audit logs is the inability of exchanging the oauthauthorizetoken for an oauthaccesstoken as we log creation of both of these objects`